File: src/main/java/com/example/cards_game/models/dtos/CreateRoomDto.java
package com.example.cards_game.models.dtos;

import jakarta.annotation.Nullable;
import lombok.Value;

import java.io.Serializable;

/**
 * DTO for {@link com.example.cards_game.models.Room}
 */
@Value
public class CreateRoomDto implements Serializable {
    int maxPlayers;
    @Nullable
    String password;
    String hostToken;
}

File: frontend/src/app/create-room-dto.ts
export class CreateRoomDto {
  maxPlayers : number;
  hostToken : string;
  password : string | null;


  constructor(maxPlayers: number, hostToken: string, password: string | null) {
    this.maxPlayers = maxPlayers;
    this.hostToken = hostToken;
    this.password = password;
  }
}
